### ISSUE TITLE
Libsndfile from the latest bottle uses @loader_path

### ISSUE THREAD

**Issue by dyfer** [Homebrew/homebrew-core#150347]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
% brew config
HOMEBREW_VERSION: 4.1.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: e57a87cfe9c15f192cd4fc6695a2b3b7acf0ddf7
Last commit: 9 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: c34af55fb3d3988c0400e9c56ccd779b1bdd49c0
Core tap last commit: 2 weeks ago
Core tap branch: master
Core tap JSON: 09 Oct 21:09 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.37.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 12.6.9-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.0
Rosetta 2: false

% brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  clang-format@8

Warning: You have the following deprecated, official taps tapped:
  Homebrew/homebrew-cask-drivers
Untap them with `brew untap`.

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libPDFNetC.dylib
  /usr/local/lib/libjack.0.dylib
  /usr/local/lib/libjacknet.0.dylib
  /usr/local/lib/libjackserver.0.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/jack/control.h
  /usr/local/include/jack/intclient.h
  /usr/local/include/jack/jack.h
  /usr/local/include/jack/jslist.h
  /usr/local/include/jack/metadata.h
  /usr/local/include/jack/midiport.h
  /usr/local/include/jack/net.h
  /usr/local/include/jack/ringbuffer.h
  /usr/local/include/jack/session.h
  /usr/local/include/jack/statistics.h
  /usr/local/include/jack/systemdeps.h
  /usr/local/include/jack/thread.h
  /usr/local/include/jack/transport.h
  /usr/local/include/jack/types.h
  /usr/local/include/jack/uuid.h
  /usr/local/include/jack/weakjack.h
  /usr/local/include/jack/weakmacros.h

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/jack.pc

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /opt/homebrew/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/opt/homebrew/sbin:$PATH"' >> ~/.zshrc

Warning: Your Xcode (14.0) is outdated.
Please update to Xcode 14.2 (or delete it).
Xcode can be updated from the App Store.
```

### Verification

- [ ] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I was locally building SuperCollider, which used homebrew's libsndfile. The build process assembles the .app bundle that includes all 3rd-party libraries (including homebrew's libsndfile) and the resulting bundle was missing libsndfile's dependencies. Upon further investigation, it turned out that libsndfile's paths to shared libraries were using `@loader_path` prefix, which made it impossible to use by SuperCollider's bundle assemble mechanism. I just wanted to point out that this was working fine up until about a month ago, probably until https://github.com/Homebrew/homebrew-core/pull/143549.
When I reinstalled libsndfile from source, the paths seem to be correct.

### What happened (include all command output)?

I installed libsndfile with `brew install libsndfile`. The library has the following dependency paths:
```
% otool -L `brew --prefix libsndfile`/lib/libsndfile.1.dylib
/opt/homebrew/opt/libsndfile/lib/libsndfile.1.dylib:
	/opt/homebrew/opt/libsndfile/lib/libsndfile.1.dylib (compatibility version 2.0.0, current version 2.37.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	@loader_path/../../../../opt/libogg/lib/libogg.0.dylib (compatibility version 0.0.0, current version 0.8.5)
	@loader_path/../../../../opt/libvorbis/lib/libvorbisenc.2.dylib (compatibility version 3.0.0, current version 3.12.0)
	@loader_path/../../../../opt/flac/lib/libFLAC.12.dylib (compatibility version 14.0.0, current version 14.0.0)
	@loader_path/../../../../opt/opus/lib/libopus.0.dylib (compatibility version 10.0.0, current version 10.0.0)
	@loader_path/../../../../opt/mpg123/lib/libmpg123.0.dylib (compatibility version 48.0.0, current version 48.0.0)
	@loader_path/../../../../opt/lame/lib/libmp3lame.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	@loader_path/../../../../opt/libvorbis/lib/libvorbis.0.dylib (compatibility version 5.0.0, current version 5.9.0)
```

### What did you expect to happen?

I expected the dependency paths to be full paths, like:
```
% otool -L `brew --prefix libsndfile`/lib/libsndfile.1.dylib
/opt/homebrew/opt/libsndfile/lib/libsndfile.1.dylib:
	/opt/homebrew/opt/libsndfile/lib/libsndfile.1.dylib (compatibility version 2.0.0, current version 2.37.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	/opt/homebrew/opt/libogg/lib/libogg.0.dylib (compatibility version 0.0.0, current version 0.8.5)
	/opt/homebrew/opt/libvorbis/lib/libvorbisenc.2.dylib (compatibility version 3.0.0, current version 3.12.0)
	/opt/homebrew/opt/flac/lib/libFLAC.12.dylib (compatibility version 14.0.0, current version 14.0.0)
	/opt/homebrew/opt/opus/lib/libopus.0.dylib (compatibility version 10.0.0, current version 10.0.0)
	/opt/homebrew/opt/mpg123/lib/libmpg123.0.dylib (compatibility version 49.0.0, current version 49.0.0)
	/opt/homebrew/opt/lame/lib/libmp3lame.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/opt/homebrew/opt/libvorbis/lib/libvorbis.0.dylib (compatibility version 5.0.0, current version 5.9.0)
```

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
% brew install libsndfile
(...)
% otool -L `brew --prefix libsndfile`/lib/libsndfile.1.dylib
/opt/homebrew/opt/libsndfile/lib/libsndfile.1.dylib:
	/opt/homebrew/opt/libsndfile/lib/libsndfile.1.dylib (compatibility version 2.0.0, current version 2.37.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	@loader_path/../../../../opt/libogg/lib/libogg.0.dylib (compatibility version 0.0.0, current version 0.8.5)
	@loader_path/../../../../opt/libvorbis/lib/libvorbisenc.2.dylib (compatibility version 3.0.0, current version 3.12.0)
	@loader_path/../../../../opt/flac/lib/libFLAC.12.dylib (compatibility version 14.0.0, current version 14.0.0)
	@loader_path/../../../../opt/opus/lib/libopus.0.dylib (compatibility version 10.0.0, current version 10.0.0)
	@loader_path/../../../../opt/mpg123/lib/libmpg123.0.dylib (compatibility version 48.0.0, current version 48.0.0)
	@loader_path/../../../../opt/lame/lib/libmp3lame.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	@loader_path/../../../../opt/libvorbis/lib/libvorbis.0.dylib (compatibility version 5.0.0, current version 5.9.0)

% brew reinstall libsndfile --build-from-source
(...)
% otool -L `brew --prefix libsndfile`/lib/libsndfile.1.dylib
/opt/homebrew/opt/libsndfile/lib/libsndfile.1.dylib:
	/opt/homebrew/opt/libsndfile/lib/libsndfile.1.dylib (compatibility version 2.0.0, current version 2.37.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	/opt/homebrew/opt/libogg/lib/libogg.0.dylib (compatibility version 0.0.0, current version 0.8.5)
	/opt/homebrew/opt/libvorbis/lib/libvorbisenc.2.dylib (compatibility version 3.0.0, current version 3.12.0)
	/opt/homebrew/opt/flac/lib/libFLAC.12.dylib (compatibility version 14.0.0, current version 14.0.0)
	/opt/homebrew/opt/opus/lib/libopus.0.dylib (compatibility version 10.0.0, current version 10.0.0)
	/opt/homebrew/opt/mpg123/lib/libmpg123.0.dylib (compatibility version 49.0.0, current version 49.0.0)
	/opt/homebrew/opt/lame/lib/libmp3lame.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/opt/homebrew/opt/libvorbis/lib/libvorbis.0.dylib (compatibility version 5.0.0, current version 5.9.0)
```

**Comment by Bo98** [comment]
This isn't really a Homebrew bug - it's a `macdeployqt` bug. There's nothing wrong with `@loader_path` on its own and you should never expect Homebrew to ship libraries with fully qualified paths (or when copying _any_ arbitrary third party library really). In fact. We ship many with `@rpath` too where the various upstreams prefer them.

`macdeployqt` has a long history of being very buggy and not reliable: https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue+macdeployqt+is%3Aclosed

With that said we've disabled `@loader_path` for now due to the Qt issues, so we could rebuild `libsndfile` here to fix things for now.

**Comment by dyfer** [comment]
Thank you so much for clarifying. I was half expecting this to be an issue with `macdeployqt`. 
However, shouldn't a local build and the bottle create a binary with the same library paths?

**Comment by cho-m** [comment]
New bottle should be available with #151855

Should be able to get it via `brew reinstall libsndfile`:
```console
❯ brew reinstall libsndfile
==> Downloading https://ghcr.io/v2/homebrew/core/libsndfile/manifests/1.2.2-1
##################################################################################################################################### 100.0%
==> Fetching libsndfile
==> Downloading https://ghcr.io/v2/homebrew/core/libsndfile/blobs/sha256:d628c38ed7a4b7a5e05f685cc69093051bc499ea54b0ce974e0bd022e06a3e41
##################################################################################################################################### 100.0%
==> Reinstalling libsndfile
==> Pouring libsndfile--1.2.2.arm64_sonoma.bottle.1.tar.gz
🍺  /opt/homebrew/Cellar/libsndfile/1.2.2: 53 files, 1MB
==> Running `brew cleanup libsndfile`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).

❯ otool -L $(brew --prefix libsndfile)/lib/libsndfile.dylib
/opt/homebrew/opt/libsndfile/lib/libsndfile.dylib:
	/opt/homebrew/opt/libsndfile/lib/libsndfile.1.dylib (compatibility version 2.0.0, current version 2.37.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1336.0.0)
	/opt/homebrew/opt/libogg/lib/libogg.0.dylib (compatibility version 0.0.0, current version 0.8.5)
	/opt/homebrew/opt/libvorbis/lib/libvorbisenc.2.dylib (compatibility version 3.0.0, current version 3.12.0)
	/opt/homebrew/opt/flac/lib/libFLAC.12.dylib (compatibility version 14.0.0, current version 14.0.0)
	/opt/homebrew/opt/opus/lib/libopus.0.dylib (compatibility version 10.0.0, current version 10.0.0)
	/opt/homebrew/opt/mpg123/lib/libmpg123.0.dylib (compatibility version 49.0.0, current version 49.1.0)
	/opt/homebrew/opt/lame/lib/libmp3lame.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/opt/homebrew/opt/libvorbis/lib/libvorbis.0.dylib (compatibility version 5.0.0, current version 5.9.0)
```

Closing as resolved. You can add comment if still have issue.

**Comment by dyfer** [comment]
Thank you!
